src/Microsoft.ServiceFabric.Services.Queryable/Controller/QueryableController.cs

```diff
@@ -70,15 +75,113 @@ public abstract class QueryableController : ApiController
 			}
 		}
 
+		protected async Task<IHttpActionResult> DeleteAsync(string application, string service, string collection,
+			ValueViewModel[] obj)
+		{
+			var serviceUri = GetServiceUri(application, service);
+			try
+			{
+
+				bool[][] results = new bool[obj.Length][];
+				for (int i = 0; i < obj.Length; i++)
+				{
+					//Serialize the key from the json body and put it into a string.
+					string keyquoted = JsonConvert.SerializeObject(obj[i].Key,
+						new JsonSerializerSettings {StringEscapeHandling = StringEscapeHandling.EscapeNonAscii});
+
+					//Fetch partition proxy.
+					var proxy = await GetServiceProxyForPartitionAsync<IQueryableService>(serviceUri, obj[i].PartitionId)
+						.ConfigureAwait(false);
+					//Perform delete opration.
+					results[i] = await Task.WhenAll(proxy.Select(p => p.DeleteAsync(collection, keyquoted)))
+						.ConfigureAwait(false);
+
+				}
+				return Ok(results);
+			}
+			catch (Exception e)
+			{
+				return HandleException(e, serviceUri);
+			}
+		}
+
+
+		protected async Task<IHttpActionResult> AddAsync(string application, string service, string collection,
+			ValueViewModel[] obj)
+		{
+			var serviceUri = GetServiceUri(application, service);
+			try
+			{
+				bool[] results = new bool[obj.Length];
+				for (int i = 0; i < obj.Length; i++)
+				{
+					//Serialize the key from the json body and put it into a string.
+					string keyquoted = JsonConvert.SerializeObject(obj[i].Key,
+						new JsonSerializerSettings {StringEscapeHandling = StringEscapeHandling.EscapeNonAscii});
+					//Serialize the value from the json body and put it into a string.
+					string valuequoted = JsonConvert.SerializeObject(obj[i].Value,
+						new JsonSerializerSettings {StringEscapeHandling = StringEscapeHandling.EscapeNonAscii});
+					//Fetch the proxy
+					var proxy = await GetServiceProxyForAddAsync<IQueryableService>(serviceUri, obj[i].PartitionId)
+						.ConfigureAwait(false);
+
+					results[i] = await proxy.AddAsync(collection, keyquoted, valuequoted).ConfigureAwait(false);
```

Comment:
Instead of a for loop, you can map over the input creating new tasks, then await all of the tasks at the same time letting them run in parallel.